packages/babel-helpers/src/helpers/applyDecs2311.ts

```diff
@@ -508,31 +496,12 @@ export default /* @no-mangle */ function applyDecs2311(
       var isField = kind === PROP_KIND.FIELD;
       var isAccessor = kind === PROP_KIND.ACCESSOR;
 
-      var key = name + "/" + isStatic;
-
-      if (!isField && !isPrivate) {
-        var existingKind = existingNonFields.get(key);
-
-        if (
-          existingKind === true ||
-          (existingKind === PROP_KIND.GETTER && kind !== PROP_KIND.SETTER) ||
-          (existingKind === PROP_KIND.SETTER && kind !== PROP_KIND.GETTER)
-        ) {
-          throw new Error(
-            "Attempted to decorate a public method/accessor that has the same name as a previously decorated public method/accessor. This is not currently supported by the decorators plugin. Property name was: " +
-              name,
```

Comment:
What happens now when we decorate a public method/accessor with the same name of one that has been already decorated?

Reply:
I thought there would be a seemingly incorrect `ctx` like https://github.com/babel/babel/pull/16263#discussion_r1478373921,
At the same time, the decorator's operation on the first property may be reflected on the second property? Because the second property overrides the first property. I'm not sure, just guessing. :)

Reply:
Unfortunately this check accounts for 10% of the helper's size. (also after gzip)